sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConsumer.cs

```diff
@@ -425,46 +1030,108 @@ internal List<ArraySegment<byte>> ConvertLockTokensToDeliveryTags(IEnumerable<Gu
                 {
                     // TODO: throw amqpResponseMessage.ToMessagingContractException();
                 }
+                return lockedUntilUtc;
             }
             catch (Exception exception)
             {
                 // TODO: throw AmqpExceptionHelper.GetClientException(exception);
                 throw exception;
             }
+        }
 
-            return lockedUntilUtc;
+        /// <summary>
+        /// Receives a <see cref="IList{Message}"/> of deferred messages identified by <paramref name="sequenceNumbers"/>.
+        /// </summary>
+        /// <param name="sequenceNumbers">An <see cref="IEnumerable{T}"/> containing the sequence numbers to receive.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>Messages identified by sequence number are returned. Returns null if no messages are found.
+        /// Throws if the messages have not been deferred.</returns>
+        /// <seealso cref="DeferAsync"/>
+        public override async Task<IList<ServiceBusReceivedMessage>> ReceiveDeferredMessageBatchAsync(
+            IEnumerable<long> sequenceNumbers,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(AmqpReceiver));
```

Comment:
nit: In other methods we are checking AssertNotClosed `ServiceBusReceiver`. Can we update the same here?

Reply:
Updated